apps/meteor/server/services/messages/service.ts

```diff
@@ -2,17 +2,42 @@ import type { IMessageService } from '@rocket.chat/core-services';
 import { ServiceClassInternal } from '@rocket.chat/core-services';
 import type { IMessage, MessageTypesValues, IUser, IRoom } from '@rocket.chat/core-typings';
 import { Messages } from '@rocket.chat/models';
+import type BadWordsFilter from 'bad-words';
 
 import { deleteMessage } from '../../../app/lib/server/functions/deleteMessage';
 import { sendMessage } from '../../../app/lib/server/functions/sendMessage';
 import { updateMessage } from '../../../app/lib/server/functions/updateMessage';
 import { executeSendMessage } from '../../../app/lib/server/methods/sendMessage';
 import { executeSetReaction } from '../../../app/reactions/server/setReaction';
 import { settings } from '../../../app/settings/server';
+import { configureBadWords } from './hooks/badwords';
 
 export class MessageService extends ServiceClassInternal implements IMessageService {
 	protected name = 'message';
 
+	private badWordsFilter?: BadWordsFilter;
+
+	async created() {
+		await this.configureBadWords();
+	}
+
+	private async configureBadWords() {
+		if (!this.isBadWordsFilterEnabled()) {
+			return;
+		}
+
+		settings.watchMultiple(
```

Comment:
I believe if the setting to enable bad words is `false` when the service was created, this will not execute and will not listen for changes on the setting after the service was created. 😬

Reply:
good call.. let me fix that